### PR TITLE
feat: DAPPS Application Name

### PR DESCRIPTION
**Pull request by fzavalia**
Changes:
- Add DAPPS app name to enum
- `StateWithFeatures` type has a possibly undefined features property. Clients might not be implementing the reducer so it might be possible that it is undefined. By typing it accordingly, we might be able to handle it right.
- `getState` now throws a `"'features' module not implemented"` error.
- `getIsFeatureEnabled` returns false if the provided app cannot be found in the state. There might be cases where the features are still loading and the app hasn't been added.

### CODE REVIEW

File: src/modules/features/types.ts [hunk 1]
```diff
@@ -38,5 +39,7 @@ export type FeatureSagasConfig = {
 }
 
 export type StateWithFeatures = {
-  features: FeaturesState
+  // Possibly undefined because clients might not have implemented te features module into their dapps.
```
Comment (cyaiox) [inline src/modules/features/types.ts:42]: typo error.
`// Possibly undefined because clients might not have implemented the features module into their dapps.`

Comment (fzavalia) [reply src/modules/features/types.ts]: Thanks 😄

File: src/modules/features/selectors.ts [hunk 2]
```diff
@@ -41,28 +50,31 @@ export const getIsFeatureEnabled = (
...
-  if (!appFeatures) {
-    throw new Error(`Application "${app}" not found`)
+  // The app might not be defined in the store because the flags might not have been fetched yet.
+  // We suggest using isLoadingFeatureFlags and hasLoadedInitialFlags to handle this first.
+  if (!appFlags) {
+    return false
   }
 
-  return !!appFeatures.flags[`${app}-${feature}`]
+  return !!appFlags.flags[`${app}-${feature}`] ?? false
```
Comment (cyaiox) [inline src/modules/features/selectors.ts:69]: the double not operator will return `false` if the value is `null || undefined || false`

Comment (fzavalia) [reply src/modules/features/selectors.ts:69]: That is the expected result

Comment (juanmahidalgo) [inline src/modules/features/selectors.ts:69]:
```suggestion
  return !!appFlags.flags[`${app}-${feature}`]
```
`!!undefined` would return `false`, right? if so, I think this `??` is redundant

File: src/modules/features/selectors.ts [hunk 3]
```diff
@@ -41,28 +50,31 @@ export const getIsFeatureEnabled = (
...
-  const features = getData(state)
-  const appFeatures: ApplicationFeatures | undefined = features[app]
+  const appFlags = getData(state)[app]
 
-  if (!appFeatures) {
-    throw new Error(`Application "${app}" not found`)
+  // The app might not be defined in the store because the flags might not have been fetched yet.
+  // We suggest using isLoadingFeatureFlags and hasLoadedInitialFlags to handle this first.
+  if (!appFlags) {
+    return false
```
Comment (juanmahidalgo) [inline src/modules/features/selectors.ts:66]: quick question! what's the motivation behind this change? returning `false` instead of throwing. Is there any scenario were we prefer that?

File: src/modules/features/selectors.ts (hunk 2 above)
Comment (fzavalia) [reply src/modules/features/selectors.ts:69]: Ohh you are right, was trying to see if `?? false` was better than `!!` and ended up using both 🙃 Will remove one. @cyaiox that's probably what you meant as well.

**Review by cyaiox — APPROVED** [review]
🚀
